### ISSUE TITLE
[Flaky Test] Create, Hibernate, Wake up and Delete Shoot e2e test is flaky

### ISSUE THREAD

**Issue by plkokanov** [gardener/gardener#6756]
**How to categorize this issue?**

/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:
The Create, Hibernate, Wake up and Delete Shoot e2e test failed with timeout during the `Wake Up` step.

**CI link**:
https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6748/pull-gardener-e2e-kind/1575081780585697280

**Reason for failure**:
It seems that waking up the shoot cluster failed with timeout while waiting for the `kube-system/blackbox-exporter` deployment to become ready:
```
    {"level":"info","ts":"2022-09-28T11:46:56.280Z","logger":"test","msg":"Last Operation","shoot":"garden-local/e2e-wake-up","lastOperation":"&LastOperation{Description:Shoot cluster has been successfully reconciled.,LastUpdateTime:2022-09-28 11:45:31 +0000 UTC,Progress:100,State:Succeeded,Type:Reconcile,}"}
    {"level":"info","ts":"2022-09-28T11:47:32.611Z","logger":"test","msg":"Shoot is not yet reconciled","shoot":"garden-local/e2e-wake-up","reason":"condition type SystemComponentsHealthy is not true yet, had message Deployment \"kube-system/blackbox-exporter\" is unhealthy: condition \"Available\" has invalid status False (expected True) due to MinimumReplicasUnavailable: Deployment does not have minimum availability. with reason DeploymentUnhealthy"}
    {"level":"info","ts":"2022-09-28T11:47:32.611Z","logger":"test","msg":"Last Operation","shoot":"garden-local/e2e-wake-up","lastOperation":"&LastOperation{Description:Shoot cluster has been successfully reconciled.,LastUpdateTime:2022-09-28 11:45:31 +0000 UTC,Progress:100,State:Succeeded,Type:Reconcile,}"}
    {"level":"info","ts":"2022-09-28T11:48:02.616Z","logger":"test","msg":"Shoot is not yet reconciled","shoot":"garden-local/e2e-wake-up","reason":"condition type SystemComponentsHealthy is not true yet, had message Deployment \"kube-system/blackbox-exporter\" is unhealthy: condition \"Available\" has invalid status False (expected True) due to MinimumReplicasUnavailable: Deployment does not have minimum availability. with reason DeploymentUnhealthy"}
```
**Anything else we need to know**:

**Comment by rfranzke** [comment]
I've checked above occurrence. Results:

- Shoot creation and hibernation succeeded, but the wake-up step failed because `blackbox-exporter` was not becoming healthy.
- Shoot was woken up at 11:38:25 and the test failed after 10min at 11:48:25 according to the [configured timeout](https://github.com/gardener/gardener/blob/66b1975adf5d6656930d1273c150792028646ad5/test/e2e/shoot/create_hibernate_wakeup_delete.go#L44)
- The first logs of the two `blackbox-exporter` pods start at 11:48:37, so just a few seconds later after the test timed out.
- I've checked some successful tests: Typically, the wake-up procedure needs ~`6m`. In above failure, I think the reason for the delay/longer duration was that the machine pod needed ~`4m` to come up again/become healthy. Usually, this takes ~`1m`.
- In the events, I saw that the readiness probe timed out a few minutes after the pod was created, however it's not clear why.

TL;DR: I don't think we have general race conditions with this test. Maybe the kind pod was overloaded, maybe the Prow cluster had some turbulences affecting the test pods.

For now, the only improvement I'm suggesting is to increase the timeout for the wake-up step to the same value like for the creation (`10m` vs. `15m`) since wake-up is basically the same like creation or reconciliation, hence it makes sense to use the same timeout.

Otherwise, I'd close this (this anyways only happened once) and see whether we have to revisit it again in the future.
